### PR TITLE
Add a GOVERNANCE.md document

### PR DESCRIPTION
**Pull request by criccomini**
Adding a governance document based on [Vespa's](https://github.com/vespa-engine/vespa/blob/master/GOVERNANCE.md).

I tweaked the following things:

- Make all committers board members for now.
- Add reference to slatedb-pulumi for the official committer list.
- Cleaned some grammar.
- Added a carveout to allow committers to merge small PRs without approval.

### CODE REVIEW

File: GOVERNANCE.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+request by clicking "approve", by making a comment indicating agreement, or by simply
+merging it. SlateDB Committers may merge their own PRs without approval from another committer
+if the PR is small, low-risk, and uncontroversial. This is left to the discretion of the
+committer.
+
+The SlateDB committers are listed in [slatedb-pulumi](https://github.com/slatedb/slatedb-pulumi/blob/main/__main__.py).
+
+# The SlateDB Board
+
+The __SlateDB Board__ has final authority over all decisions regarding the SlateDB project,
```
Comment (rodesai) [inline GOVERNANCE.md:42]: should we specify how we decide (e.g. simple majority)?

File: GOVERNANCE.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+# SlateDB Committers
+
+A __SlateDB Committer__ is able to merge pull requests into the SlateDB repositories under
+[https://github.com/slatedb](https://github.com/slatedb), including into
+the master branch that is released to production.
+
+To become a SlateDB Committer, start making PRs that are approved. When a sufficient
+trust is built through working on SlateDB this way, any current SlateDB Committer can nominate
+you to the board for the committer role.
```
Comment (rodesai) [inline GOVERNANCE.md:30]: how should we decide on adding a committer? the board votes (I think this is pretty typical)?

Comment (rodesai) [reply GOVERNANCE.md:30]: nvm see this is addressed below

File: GOVERNANCE.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+Every pull request should be approved by a SlateDB Committer other than the author of the
```
Comment (rodesai) [inline GOVERNANCE.md:32]: I think we should have a higher standard for RFCs - maybe 2 committer approvals and no objecting votes from committers?

**Comment by criccomini** [comment]
Updated GOVERNANCE.md and RELEASE.md with:

- Exclude slatedb-go
- Add releases to governance
- Exapand release doc with cadence, proposal, and approval
- Add RFC to governance
- Split board from committers

**Review by vigneshc — APPROVED** [review]

**Review by rodesai — APPROVED** [review]

**Review by flaneur2020 — APPROVED** [review]

**Review by paulgb — APPROVED** [review]
LGTM!

**Pull request merged**